### PR TITLE
Merge TRRFile and XTCFile into a single XDRFile class

### PR DESCRIPTION
**Pull request by Luthaf**
There was a lot of code repetition between these two classes

### CODE REVIEW

**Comment by Luthaf** [comment]
@ezavod would you have time to give this a look?

File: include/chemfiles/files/XDRFile.hpp [hunk 1]
```diff
@@ -1,26 +1,41 @@
...
+        /// XTC trajectory
+        XTC,
+        /// TRR trajectory
+        TRR,
+    };
+
+    XDRFile(Variants variant, std::string path, File::Mode mode);
+    ~XDRFile() override;
+
+    XDRFile(XDRFile&& other) noexcept: File("", File::READ, File::DEFAULT) {
```
Comment (ezavod) [inline include/chemfiles/files/XDRFile.hpp]: To be nitpicky: missing space before colon `noexcept : File` =)

File: include/chemfiles/formats/XTC.hpp [hunk 2]
```diff
@@ -19,13 +19,8 @@ class XTCFormat final : public Format {
...
   private:
-    void set_positions(const std::vector<float>& x, Frame& frame);
-    void get_positions(std::vector<float>& x, const Frame& frame);
-    void set_cell(matrix box, Frame& frame);
-    void get_cell(matrix box, const Frame& frame);
-
-    /// Reference to the associated file
-    XTCFile xtc_;
+      /// Associated XDR file
+      XDRFile file_;
```
Comment (ezavod) [inline include/chemfiles/formats/XTC.hpp]: Indentation too far?

File: src/files/XDRFile.cpp [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        read_header(this->path().c_str(), &natoms_, &nframes_, &offsets_);
+    }
+
+    handle_ = xdrfile_open(this->path().c_str(), openmode);
+    if (handle_ == nullptr) {
+        throw file_error("could not open the file at '{}'", this->path());
+    }
+}
+
+
```
Comment (ezavod) [inline src/files/XDRFile.cpp]: Only one newline?

File: tests/CMakeLists.txt [hunk 4]
```diff
@@ -40,6 +40,7 @@ if(${CHFL_TEST_RUNNER} STREQUAL "valgrind")
         RUNNER_COMMAND
         "valgrind" "--leak-check=full" "--dsymutil=yes" "--error-exitcode=125"
         "--suppressions=${CMAKE_CURRENT_SOURCE_DIR}/valgrind.supp"
+        "--track-origins=yes" "--leak-check=full" "--show-leak-kinds=all"
```
Comment (ezavod) [inline tests/CMakeLists.txt]: Should `--leak-check=full` be there twice?

Comment (Luthaf) [reply tests/CMakeLists.txt]: whoops, that was debug code that should not have been included ...

**Pull request merged**